app/models/work.rb

```diff
@@ -712,28 +693,24 @@ def publish_precurated_files
       # An error is raised if there are no files to be moved
       raise(StandardError, "Attempting to publish a Work without attached uploads for #{s3_object_key}") if pre_curation_uploads.empty? && post_curation_uploads.empty?
 
-      # We need to explicitly access to post-curation services here.  Lets explicitly create it so the state of the work does not have any impact
+      # We need to explicitly access to post-curation services here.
+      # Lets explicitly create it so the state of the work does not have any impact.
       s3_post_curation_query_service = S3QueryService.new(self, false)
 
-      # This migrates pre-curation S3 Objects to the post-curation S3 Bucket
-      transferred = []
-
       s3_dir = find_post_curation_s3_dir(bucket_name: s3_post_curation_query_service.bucket_name)
       raise(StandardError, "Attempting to publish a Work with an existing S3 Bucket directory for: #{s3_object_key}") unless s3_dir.nil?
 
-      pre_curation_uploads.each do |attachment|
-        next if post_curation_s3_file?(filename: attachment.filename.to_s)
+      # Copy the pre-curation S3 Objects to the post-curation S3 Bucket...
+      transferred_files = s3_post_curation_query_service.publish_files
 
-        put_post_curation_s3_object(bucket_name: s3_post_curation_query_service.bucket_name, blob: attachment.blob)
+      # ...check that the files are indeed now in the post-curation bucket...
```

Comment:
We might *not* need to check that the files were transferred since we are letting AWS do this for us, we used to do the transfer manually so the check made sense. I left the check for now since it does not really hurt and it probably won't slow things down.